src/Generator.cpp

```diff
@@ -0,0 +1,36 @@
+#include "Generator.hpp"
+#include <iostream>
+
+Lines getAllVectorPermutation_(int vSize)
+{
+    std::vector<int> row(vSize);
+    Lines result;
+
+    for(;;)
+    {
+        result.emplace_back(row);
+        
+        for (int i = row.size() - 1; ; --i)
+        {
+            if (i < 0) return result;
+
+            if (++row[i] == 2) row[i] = 0;
```

Comment:
this algorithm looks complicated. Have you considered using std::next_pemutation?

Reply:
I have consider it but had big trouble to use it in case when lexicon vector for permutation is smaller then output vector.

Reply:
Ok, if this algorithms stays like it is, please change the formatting.
Personally, I don't like when it's easy to miss what is executed if the condition is true. Could you put `row[i] = 0;` in the next line with indentation?
The same for line 14 and 17.

Reply:
done